src/macaron/dependency_analyzer/dependency_resolver.py

```diff
@@ -129,6 +130,16 @@ def add_latest_version(
         url_to_artifact: dict[str, set]
             Used to detect artifacts that have similar repos.
         """
+        if item["url"] == "" and item["version"] != "unspecified":
+            gav = f"{item['group']}:{item['name']}:{item['version']}"
```

Comment:
If `item['group']` or `item['name']` is empty, would the `gav` be valid?

Reply:
It would not. I've added additional checks.